board/app/controllers/thread_controller.php

```diff
@@ -1,107 +1,149 @@
 <?php
+
 class ThreadController extends AppController {
 
-    const MAX_THREAD_PER_PAGE = 7;
-    const MAX_COMMENT_PER_PAGE = 5;
-  
-    /*
-    * Create new thread
-    * :: - STATIC FUNCTION, can be called from the class name
-    * -> - INSTANCE, can only be called from an instance of the class.
-    */
-    public function create()
+const CREATE_THREAD = 'create';
+const CREATE_THREAD_END = 'create_end';
+
+/*
+* Create new thread
+* :: - STATIC FUNCTION, can be called from the class name
+* -> - INSTANCE, can only be called from an instance of the class.
+*
+* Everything inputted on the form (view/thread/create.php) will be gathered by this function
+*/
+
+   public function create() 
     {
         $thread = new Thread();
         $comment = new Comment();
-        $current_page = Param::get('page_next', 'create');
-      
-        switch ($current_page) {
-            case 'create':
-            break;
-      
-        case 'create_end':
-            $thread->title = Param::get('title');
-            $comment->username = Param::get('username');
-            $comment->body = Param::get('body');
- 
-            try 
-            {
-                $thread->create($comment);
-            } catch (ValidationException $e) {
-                $current_page = 'create';
-            }
-              break;
+        $current_page = Param::get('page_next', self::CREATE_THREAD);   
+                    
+        switch ($current_page) { 
+            case self::CREATE_THREAD:
+                break;
+            /*  
+            *   After the user clicked on submit, the page will be redirected to 'create_end'
+            *   From the $thread database, this will get the title.. and so on. 
+            *   after all, controllers are all about getting the inputted data.
+            *   then the data gathered here will be tranferred to view (view/thread/view.php)
+            */
+            case self::CREATE_THREAD_END:
+                $thread->title = Param::get('title'); 
+                $thread->category = Param::get('category');
+                $thread->user_id = $_SESSION['user_id'];
+                $comment->body = Param::get('body');
+               // $comment->username = Param::get('username');
```

Comment:
Done po. thank you :bow:
--> https://github.com/maragatchalian/FoodFacts/commit/5ed59d3636552bdd4f0e9ca6a2ec67be76ce0e16